tests/job-spawner-and-cleaner.py

```diff
@@ -13,13 +13,18 @@
 
 if len(sys.argv) < 2:
     raise Exception("first command line argument must be redis host")
-db = redis.Redis(host=sys.argv[1].replace(":6379",""))
+
+host = sys.argv[1].split(":")
+
+if len(host) == 2:
+    db = redis.Redis(host=host[0], port=host[1])
+else:
+    db = redis.Redis(host=host[0])
```

Comment:
What if I pass `localhost:123:456`?

Also, `host[1]` will be a string, is that allowed to be passed as `port`?